drivers/node/vsphere/vsphere.go

```diff
@@ -862,3 +862,336 @@ func (v *vsphere) RemoveNonRootDisks(n node.Node) error {
+	if len(datastores) == 0 {
+		return fmt.Errorf("no compatible datastores available for storage vMotion")
+	}
+	if len(datastores) > 1 {
+		maxAvailableSpace := int64(-1)
+		for _, ds := range datastores {
+			var dsProps mo.Datastore
+			if err := ds.Properties(ctx, ds.Reference(), []string{"summary"}, &dsProps); err == nil {
+				if available := dsProps.Summary.FreeSpace; available > maxAvailableSpace {
+					maxAvailableSpace = available
+					targetDatastore = ds
+				}
+			}
+		}
+	} else {
+		targetDatastore = datastores[0]
+	}
+
+	if targetDatastore == nil {
+		return fmt.Errorf("failed to select a target datastore")
+	}
+
+	for i := range diskLocators {
+		diskLocators[i].Datastore = targetDatastore.Reference()
+	}
+
+	relocateSpec := types.VirtualMachineRelocateSpec{
+		Disk: diskLocators,
+	}
+
+	task, err := vm.Relocate(ctx, relocateSpec, types.VirtualMachineMovePriorityDefaultPriority)
+	if err != nil {
+		return fmt.Errorf("error initiating VM relocate: %v", err)
+	}
+	return task.Wait(ctx)
+}
+
+// FindVMByName finds a virtual machine by its name.
+func (v *vsphere) FindVMByName(vmName string) (*object.VirtualMachine, error) {
+	log.Infof("VM Name is: %v", vmName)
+	finder, err := v.getVMFinder()
+	if err != nil {
+		return nil, err
+	}
+	vm, err := finder.VirtualMachine(v.ctx, vmName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find VM: %v", err)
+	}
+	return vm, nil
+}
+
+// FindDatastoreByName finds a datastore by its name.
+func (v *vsphere) FindDatastoreByName(dsName string) (*object.Datastore, error) {
+	finder, err := v.getVMFinder()
+	if err != nil {
+		return nil, err
+	}
+	ds, err := finder.Datastore(v.ctx, dsName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find datastore: %v", err)
+	}
+	return ds, nil
+}
+
+// FindVMByIP finds the vsphere Name of a VM through its Data IP Address
+func (v *vsphere) FindVMByIP(node node.Node) (*object.VirtualMachine, error) {
+	log.Infof("Searching VM by IP Addresses: %v", node.Addresses)
+	finder, err := v.getVMFinder()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get VM finder: %v", err)
+	}
+
+	vms, err := finder.VirtualMachineList(v.ctx, "*")
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve VM list: %v", err)
+	}
+
+	for _, vm := range vms {
+		var vmProps mo.VirtualMachine
+		err = vm.Properties(v.ctx, vm.Reference(), []string{"guest.net"}, &vmProps)
+		if err != nil {
+			log.Infof("failed to get properties for VM: %s, error: %v", vm.Name(), err)
+			continue
+		}
+
+		for _, net := range vmProps.Guest.Net {
+			for _, ip := range net.IpAddress {
+				for _, nodeIP := range node.Addresses {
+					if ip == nodeIP {
+						log.Infof("Found VM by IP Address %s: %s", ip, vm.Name())
+						return vm, nil
+					}
+				}
+			}
+		}
+	}
+	return nil, fmt.Errorf("no VM found with the given IP addresses: %v", node.Addresses)
+}
+
+// ValidateDatastoreUpdate validates the cloud drive configmap after Storage vmotion is done
```

Comment:
Example 
`    "[datastore-3022] pool-0-5b5bbd8766-fnxbn/pool-0-5b5bbd8766-fnxbn_9.vmdk": {
        "Type": "thin",
        "Size": 125,
        "ID": "[datastore-3022] pool-0-5b5bbd8766-fnxbn/pool-0-5b5bbd8766-fnxbn_9.vmdk",
        "PoolID": "4b0ce8c7-afe0-4a11-9e28-03a479242f24",
        "Path": "/dev/sdi",
        "Iops": 0,
        "Vpus": 0,
        "PXType": "data",
        "State": "In Use",
        "labels": {
          "datastore": "mh500-3-ds02-b"
        },
        "AttachOptions": {
          "ext": ".vmdk"
        },
        "Provisioner": "",
        "EncryptionKeyInfo": "",
        "DiskUUID": "6000c29e-ff89-7445-0195-68ad440c3320"
      }
`
Need to verify 
"DiskUUID": "6000c29e-ff89-7445-0195-68ad440c3320"
"ID": "[datastore-3022] pool-0-5b5bbd8766-fnxbn/pool-0-5b5bbd8766-fnxbn_9.vmdk",
"labels": {
          "datastore": "mh500-3-ds02-b"
        },

Reply:
Already doing this validation.